### ISSUE TITLE
fractional coarse-graining

### ISSUE THREAD

**Issue by anhhuyalex** [anhhuyalex/renormalization#6]
for k in 28 27 25 23 21 19 17 15 13 11 9 7 5 3 1; do python -u mnist_classification_wholenetwork.py ./data  --target_size  $k --num_hidden_features $j --num_train_samples 60000  --fileprefix relulr0.1_mnist_fullnet_fractional_cg_DEC04  --lr 0.1 --nonlinearity relu  --epochs 150 --upsample; done #    --wd 0.0000000000000000000000001

for k in 28 27 25 23 21 19 17 15 13 11 9 7 5 3 1; do python -u mnist_classification_wholenetwork.py ./data  --target_size  $k --num_hidden_features $j --num_train_samples 60000  --fileprefix linelr0.1_mnist_fullnet_fractional_cg_DEC04  --lr 0.1 --nonlinearity relu  --epochs 150 --upsample; done #    --wd 0.0000000000000000000000001

**Comment by anhhuyalex** [comment]
for k in 1 3 5 7 9 11 13 15 17 19 21 23 25 27 28; do python -u mnist_classification_lbfgs.py --data fashionmnist --penalty none --fileprefix multiclasslbfgs_mnist_fractional_cg_no_transform_JAN22   --target_size  $k   --num_train_samples $j     ; done # --wd 0.0000000000000000000000001  

for k in 1 3 5 7 9 11 13 15 17 19 21 23 25 27 28; do python -u mnist_classification_lbfgs.py --data fashionmnist --penalty none --fileprefix noregularize_multiclasslbfgs_mnist_fractional_cg_no_transform_JAN22   --target_size  $k   --num_train_samples $j     ; done # --wd 0.0000000000000000000000001

**Comment by anhhuyalex** [comment]
for k in 1 3 5 7 9 11 13 15 17 19 21 23 25 27 28; do python -u mnist_classification_lbfgs.py --data mnist --penalty none --fileprefix multiclasslbfgs_mnist_fractional_cg_no_transform_JAN22 --target_size $k --num_train_samples $j ; done # --wd 0.0000000000000000000000001

for k in 1 3 5 7 9 11 13 15 17 19 21 23 25 27 28; do python -u mnist_classification_lbfgs.py --data mnist --penalty none --fileprefix noregularize_multiclasslbfgs_mnist_fractional_cg_no_transform_JAN22 --target_size $k --num_train_samples $j ; done # --wd 0.0000000000000000000000001

**Comment by anhhuyalex** [comment]
Try: learning rate scheduler linearly reduce or cosine scheduler
Try: increase the # epochs
Study the distributions of image pixels before and after coarse graining

**Comment by anhhuyalex** [comment]
for k in 1 3 5 7 9 11 13 15 17 19 21 23 25 27 28; do python -u mnist_classification_scale_epochs.py ./data --fileprefix linearschedule_full_batch_no_regularize_multiclasslr$1_mnist_JAN26 --lr $1 --lr_scheduler linear --multiclass_lr  --target_size  $k -b $j --num_hidden_features $i --num_train_samples $j --epochs 500 --upsample --no_transform --wd 0.0; done # --wd 0.0000000000000000000000001 


for k in 1 3 5 7 9 11 13 15 17 19 21 23 25 27 28; do python -u mnist_classification_scale_epochs.py ./data --fileprefix cosineschedule_full_batch_no_regularize_multiclasslr$1_mnist_JAN26 --lr $1 --lr_scheduler cosine --multiclass_lr  --target_size  $k -b $j --num_hidden_features $i --num_train_samples $j --epochs 500 --upsample --no_transform --wd 0.0; done # --wd 0.0000000000000000000000001

**Comment by anhhuyalex** [comment]
num_hidden_features in [10000,20000,30000,40000,50000,60000]

python -u mnist_classification_scale_epochs.py ./data --fileprefix Adamonecyclelrschedule_full_batch_no_regularize_linear$1_mnist_FEB5 --optimizer_type adam --lr $1 --lr_scheduler OneCycleLR  --target_size  $i -b 500 --nonlinearity line --num_train_samples $j --num_hidden_features $2 --epochs 500 --upsample --no_transform --wd 0.0

**Comment by anhhuyalex** [comment]
reduce P (# of parameters) and N
try on fashionmnist

**Comment by anhhuyalex** [comment]
della
sbatch --array=0-89 hipp.sh 0.01 10000
python -u mnist_classification_scale_epochs.py ./data --fileprefix Adamonecyclelrschedule_full_batch_no_regularize_relu$1_mnist_MAR4 --optimizer_type adam --lr $1 --lr_scheduler OneCycleLR  --target_size  $i -b 500 --nonlinearity relu --num_train_samples $j --num_hidden_features $2 --epochs 500 --upsample --no_transform --wd 0.0  --randomfeatures --save_dir /scratch/qanguyen/imagenet_info_relurandom #
